ironmon_tracker/Program.lua

```diff
@@ -416,6 +441,48 @@ function Program.stepFrames()
 	Program.Frames.three_sec_update = (Program.Frames.three_sec_update - 1) % 180
 	Program.Frames.saveData = (Program.Frames.saveData - 1) % 3600
 	Program.Frames.carouselActive = Program.Frames.carouselActive + 1
+	if Program.Frames.hideEnemy > 0 then
+		Program.Frames.hideEnemy = Program.Frames.hideEnemy - 1
+	end
+
+	for _, framecounter in pairs(Program.Frames.Others or {}) do
+		if type(framecounter.step) == "function" then
+			framecounter:step()
+		end
+	end
+
+	SpriteData.updateActiveIcons()
+end
+
+-- Creates a frame counter that executes 'callFunc' after 'N=frames' emulation steps.
+-- 'scaleWithSpeedup': if true, will sync to real life time instead of emulation client frame rate / speed
+function Program.addFrameCounter(label, frames, callFunc, scaleWithSpeedup)
+	if label == nil or (frames or 0) <= 0 then return nil end
+	Program.Frames.Others[label] = {
+		framesElapsed = 0.0,
+		maxFrames = frames,
+		paused = false,
+		pause = function(self) self.paused = true end,
+		unpause = function(self) self.paused = false end,
+		step = function(self)
+			if self.paused then return end
+			-- Sync with client frame rate (turbo/unthrottle)
+			local delta = scaleWithSpeedup and (1.0 / Program.clientFpsMultiplier) or 1
+			self.framesElapsed = self.framesElapsed + delta
+			if self.framesElapsed >= self.maxFrames then
+				self.framesElapsed = 0.0
+				if type(callFunc) == "function" then
+					callFunc()
+				end
```

Comment:
with a variable name like `callFunc` I imagine the intent is *always* to pass in a function?

In which case, since this is an internal function only ever used with input arguments defined by ourselves, I feel like it would make more sense to check if `callFunc` is a function at the start of this function and if not then output some sort of error message for the dev to say "hey, this expects a function"?

My point here being that something within our control where an input argument can only be incorrect *if* a dev has incorrectly called this function **should** bring up an error and not obfuscate the incorrect usage behind a failsafe, the concern being that the latter naturally lets the dev be more lax in ways that can let bugs slip through

I'd also argue the same regarding `label` and `frames`, if the values for those are incorrectly passed in then it's the fault of the dev and they should be informed via an error message about it.

Reply:
Updated the internal documentation for this function such that it is significantly more clear how it's expected to be used, and that the `callFunc` is an optional parameter.